plan_test.go

```diff
@@ -638,7 +640,7 @@ func TestQueryPlanSupportsUnions(t *testing.T) {
         {
           "ServiceURL": "A",
           "ParentType": "Query",
-          "SelectionSet": "{ animals { ... on Dog { name __typename } ... on Cat { name __typename } ... on Snake { name __typename } } }",
```

Comment:
Might be worth duplicating this test with `DontDuplicateTypename` as a suffix with the selection set `{ animals { ... on Dog { name __typename } ... on Cat { name __typename } ... on Snake { name __typename }  __typename } }` to show that `__typename` does not get duplicated.